### PR TITLE
Persistent storage

### PR DESCRIPTION
**Pull request by gccurtis**
Comments stored persistently in datastore

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/comments")
+public class CommentServlet extends HttpServlet {
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  // private ArrayList<String> comments = new ArrayList<String>();
```
Comment (ohfunc) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java]: Please remove or uncomment. Same for below.

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+    Query query = new Query("Comment").addSort("timestamp",Query.SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    ArrayList<String> comments = new ArrayList<String>();
+    for(Entity entity : results.asIterable()){
+	    comments.add((String) entity.getProperty("message"));
+    }
+    String json = gson.toJson(comments);
+    response.setContentType("application/json");
```
Comment (ohfunc) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java]: Do you need a semicolon after the content type?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -16,13 +16,61 @@
...
+function removeChildren(node){
+  const children = node.childNodes;
+  while(node.hasChildNodes()){
+    node.removeChild(children[0]);
+  }
+  console.log(node.childNodes.length);
+}
+
+function deleteComment(id){
+  //const myHeaders = new Headers();
```
Comment (Howon) [inline portfolio/src/main/webapp/script.js:42]: Remove comments/logging for pull requests

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -16,13 +16,61 @@
...
+  fetch('/data2?name='+name).then(response => response.text()).then((data) => {
+    document.getElementById('data-display').innerText = data;
+  });
+}
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+function removeChildren(node){
+  const children = node.childNodes;
+  while(node.hasChildNodes()){
```
Comment (Howon) [inline portfolio/src/main/webapp/script.js:35]: let's add some white spaces
```
while (node.haschildnodes()) {
```

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -16,13 +16,61 @@
...
+  fetch('/data').then(response => response.json()).then((data) => {
+    const greeting = data[Math.floor(Math.random()*data.length)];
+    const greetingContainer = document.getElementById('greeting-container');
+    greetingContainer.innerText = greeting;
+  });
+}
+
+function getData(){
+  const name = document.getElementById('text-input').value; 
+  fetch('/data2?name='+name).then(response => response.text()).then((data) => {
```
Comment (Howon) [inline portfolio/src/main/webapp/script.js]: String interpolation here

**Review by Howon — APPROVED** [review]

**Pull request merged**